[FEATURE] support for ILIAS 10

chfsx opened the pull request:
Hi @okaufman 

This "PR" is a version of the plugin for ILIAS 10. please note that this PR cannot be merged but should result ina new branch (e.g. ilias10).

chfsx committed:
[FEATURE] support for ILIAS 10

# Conflicts:
#	classes/class.ilSelfEvaluationPlugin.php

chfsx committed:
[FIX] JS errors

chfsx committed:
[FIX] several path issues

# Conflicts:
#	src/UIHelper/MatrixFieldInputGUI.php

chfsx committed:
[FIX] block property declaration

chfsx committed:
[FIX] rebuilt JS with vanilla

ILIASLM commented:
Nice! Dann ist das ja schon auf gutem Wege. Auch wenn das Merging noch nicht stattgefunden hat.

chfsx committed:
[FIX] SUPPORT-17784: Show icon for incomplete dataset in results table

Replace the broken-image placeholder shown in the "Abgeschlossen" column
of the dataset results table with the standard ILIAS icon. The previous
code used $plugin->getDirectory() (absolute filesystem path) as an <img>
src, which the browser cannot resolve, so incomplete rows rendered with
the broken-image icon instead of empty.png.

Use ilUtil::getImagePath() for both states and switch the incomplete
state to standard/icon_not_ok.svg so users see a clear cross/check pair.
